community/cypher/cypher-compiler-2.3/src/test/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResultTest.scala

```diff
@@ -131,6 +130,21 @@ class CompiledExecutionResultTest extends CypherFunSuite {
     intercept[IllegalStateException](result.javaIterator.hasNext)
   }
 
+  test("close should work after result is consumed") {
+    // given
+    val result = newCompiledExecutionResult(javaMap("a" -> "1", "b" -> "2"))
+
+    // when
+    result.accept(new ResultVisitor[Exception] {
+      override def visit(row: ResultRow): Boolean = {
+        true
+      }
+    })
+
+    // then
+    result.close()
```

Comment:
can you say that? perhaps with `// then doesn't blow up`.

Moreover I would say that `result.close()` is your "when" rather than your "then".